### ISSUE TITLE
AWS::CodeDeploy::BlueGreen fails when Target Groups protocol is HTTP2/gRPC

### ISSUE THREAD

**Issue by angel-eb** [aws-cloudformation/cloudformation-coverage-roadmap#1115]
### Name of the resource

Other

### Resource Name

AWS::CodeDeploy::BlueGreen

### Issue Description

I'm trying to setup BlueGreen deployments for a gRPC service in ECS with `AWS::CodeDeploy::BlueGreen`.

Setting `AWS::CodeDeploy::BlueGreen` in CloudFormation works as expected following [the documentation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/blue-green.html). The service is initially deployed using the initial Task Definition and in the initial setup. CodeDeploy is not used. The particularity about the setup is that both the blue and green target groups has `ProtocolVersion` attribute set to `GRPC`.

The issue comes when there is an update in the TaskDefinition that needs to be deployed through CodeDeploy.

This is the `cdk diff` when the task is updated to a new image:

```
[~] AWS::ECS::TaskDefinition BlueGreenGRPC/FargateTask/FargateTaskDef BlueGreenGRPCFargateTaskFargateTaskDef37AF0EE6 replace
 └─ [~] ContainerDefinitions (requires replacement)
     └─ @@ -18,7 +18,7 @@
        [ ]       {
        [ ]         "Ref": "AWS::URLSuffix"
        [ ]       },
        [-]       "/grpcservice:d79284"
        [+]       "/grpcservice:c15139"
        [ ]     ]
        [ ]   ]
        [ ] },
[~] AWS::ECS::TaskSet BlueGreenGRPC/FargateTask/BlueTaskSet BlueGreenGRPCFargateTaskBlueTaskSet1D69FA57 replace
 └─ [~] TaskDefinition (requires replacement)
     └─ [~] .Ref:
         ├─ [-] BlueGreenGRPCFargateTaskFargateTaskDef37AF0EE6
         └─ [+] BlueGreenGRPCFargateTaskFargateTaskDef37AF0EE6 (replaced)

```

When I run the Change Set in CloudFormation to review what is actually updated after applying the BlueGreen hook, I can see resources like the ELB, TaskSet and TaskDefinitions involved being updated, which is expected, but then I also see the target groups being replaced.

In the detail of the template to be deployed I see this:

```json
    "BlueGreenGRPCTargetGroupBlueA740429C": {
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
      "Metadata": {
        "aws:cdk:path": "BlueGreenStack/BlueGreenGRPC/TargetGroupBlue/Resource"
      },
      "Properties": {
        "Protocol": "HTTP",
        "Port": 50051,
        "TargetType": "ip",
        "VpcId": "vpc-xxxxxxxxxxx",
        "TargetGroupAttributes": [
          {
            "Value": "false",
            "Key": "stickiness.enabled"
          }
        ]
      }
    },
```

In the initial template, this target group looks like this:

```json
    "BlueGreenGRPCTargetGroupBlueA740429C": {
      "Type": "AWS::ElasticLoadBalancingV2::TargetGroup",
      "Properties": {
        "Port": 50051,
        "Protocol": "HTTP",
        "ProtocolVersion": "GRPC",
        "TargetGroupAttributes": [
          {
            "Key": "stickiness.enabled",
            "Value": "false"
          }
        ],
        "TargetType": "ip",
        "VpcId": "vpc-xxxxxxxxxxx"
      },
      "Metadata": {
        "aws:cdk:path": "BlueGreenStack/BlueGreenGRPC/TargetGroupBlue/Resource"
      }
    },
```

Please note that the hook is removing/ignoring the `ProtocolVersion` attribute, therefore forcing the recreation of the target group. Furthermore, this makes the CodeDeploy fail, because without a ProtocolVersion, the target group is created with HTTP1 as `ProtocolVersion`, which is incompatible with the gRPC Service.

### Expected Behavior

`AWS::CodeDeploy::BlueGreen` hook should not ignore the `ProtocolVersion` used in the green/blue target groups.

### Observed Behavior

`AWS::CodeDeploy::BlueGreen` hook ignores the `ProtocolVersion` used in the green/blue target groups.

### Test Cases

Detailed in the Issue Description.

**Comment by AnandarajuCS** [comment]
@angel-eb @cuongvo We did a test to perform ECS BlueGreen deployment using CodeDeploy through CloudFormation and used the `ProtocolVersion` parameter in the `AWS::ElasticLoadBalancingV2::TargetGroup` in the template. The `ProtocolVersion` parameter was accepted while CFn initiated the CodeDeploy BG deployment. For the interest of time we only set the `ProtocolVersion` to `HTTP1` while testing. We believe the fix should work for any type of `ProtocolVersion` since the issue was that `ProtocolVersion` was not supported initially. Feel free to test it with `HTTP2` or `GRPC`. We will resolve this issue. Feel free to open an issue if you see any other problems.
